modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module

```diff
@@ -433,10 +505,46 @@ function manatime_feature_timesheets_add_row_form_submit(&$form, &$form_state) {
   // Add new row.
   $remove_link_row = manatime_feature_timesheet_make_link_remove_row($node, $task->getIdentifier());
   $timesheet->addRecord($project->label(), $task->label(), array(), 0, $remove_link_row);
+  $records = $timesheet->getRecords();
+  $variables = array(
+    'timesheet_id' => $node->nid,
+    'records' => $records,
+    'add_row_link' => manatime_feature_timesheet_make_link_add_row($timesheet->getNid()),
+  );
+  $markup = theme('timesheet-view', $variables);
+  // Tell the browser to close the modal.
+  $form_state['ajax_commands'][] = ctools_modal_command_dismiss();
+  // Tell the browser to replace the old link with the new one.
+  $form_state['ajax_commands'][] = ajax_command_replace('#time-entries-main', $markup);
+}
 
+/**
+ * Drupal form submit handler - Add time entry.
+ */
+function manatime_feature_timesheets_add_time_entry_form_submit(&$form, &$form_state) {
+  $node = $form_state['build_info']['args'][0];
```

Comment:
In order to avoid getting values from $form_state['build_info']['args'][0] (which could be harder to maintain) and get them from $form_state['values']['index_name'] (2/2)